src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp

```diff
@@ -22,11 +22,35 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
         VectorDims present_v_dims = input_shapes.back().get();
         const auto& beam_idx_dims = input_shapes.end()[-3].get();
         const auto& permute_axes = m_config.permute_axes;
-
         if (permute_axes.empty()) {
             // [B, H, L, S]
             present_v_dims[0] = beam_idx_dims[0];
             present_v_dims[2] += query_dims[2];
+            if (input_shapes.size() == 7 && !m_config.is_causal) {
+                const auto& attn_mask_dims = input_shapes[3].get();
```

Comment:
From the [spec](https://docs.openvino.ai/2025/documentation/openvino-ir-format/operation-sets/operation-specs/sequence/scaled-dot-product-attention.html), the dimension of attention mask may be 2+.

Reply:
fix, and add more test case